smolyay/basis.py

```diff
@@ -87,24 +127,14 @@ class ChebyshevFirstKind(BasisFunction):
 
     def __init__(self, n):
         super().__init__()
-        self._n = n
-        self._derivative_polynomial = None
-        if n > 0:
-            self._points = [-numpy.cos(numpy.pi*i/n) for i in range(n+1)]
-        else:
-            self._points = [0]
+        self.n = n
```

Comment:
You might want to do this to force a copy and also make sure we have an int
```suggestion
        self.n = int(n)
```
Sorry to backtrack on this, but this actually means that `n` should probably be a property with a getter and a setter. The setter should do this int cast.